lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -71,6 +71,7 @@ def serializable_hash_for_single_resource(options)
         end
 
         def resource_identifier_type_for(serializer)
+          return serializer._type if serializer._type.present?
```

Comment:
Since the default is nil, anything else is user entered.  This basically just guards against an empty string, but since they want that, let'em :)

I'd prefer `return serializer._type if serializer._type`... since it doesn't have to go all the way up the method call stack... https://github.com/rails/rails/blob/4-0-stable/activesupport/lib/active_support/core_ext/object/blank.rb  

but if you really like present?, not a blocker.  I incline to avoid monkeypatching